frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -587,18 +605,87 @@ impl<T: Config> Pallet<T> {
 						Ok(now)
 					}
 				}
-			});
+			},
+		);
 
 		match mutate_stat {
 			// all good
 			Ok(Ok(_)) => Ok(()),
 			// failed to write.
-			Ok(Err(_)) => Err(MinerError::Lock("failed to write to offchain db.")),
+			Ok(Err(_)) => Err(MinerError::Lock("failed to write to offchain db [repeat].")),
 			// fork etc.
 			Err(why) => Err(MinerError::Lock(why)),
 		}
 	}
 
+	/// Ensure a new offchain worker is not being executed at this point in time, by checking
+	/// `OFFCHAIN_RUNNING` storage.
+	///
+	/// If `OK(_)` is returned, `true` is written to `OFFCHAIN_RUNNING`, which in turn prevents any
+	/// further call to this function to succeed.
+	///
+	/// It is up to the call site, and ABSOLUTELY CRUCIAL to make sure `release_offchain_lock` is
+	/// called appropriately to ensure starvation does not occur.
```

Comment:
TBH this might be better ensured by implementing a guard structure with `Drop` implementation, now you have to manually check all possible cases and we don't cater for `panic` scenario correctly.